zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/util/SqlSplitter.java

```diff
@@ -102,15 +102,15 @@ public SqlSplitter(String... additionalSingleCommentPrefixList) {
         multiLineComment = false;
       }
 
-      if (character == '\'') {
+      if (character == '\'' && (!singleLineComment && !multiLineComment)) {
```

Comment:
As shown below, moving "!" out of parentheses will make it easier to understand the conditions.
```
if (character == '\'' && !(singleLineComment || multiLineComment)) {
```

Reply:
Addressed